api/routers/edr.py

```diff
@@ -58,13 +59,13 @@ async def get_locations(bbox: str = Query(..., example="5.0,52.0,6.0,52.1")) ->
 @router.get(
     "/locations/{location_id}",
     tags=["Collection data queries"],
-    response_model=Coverage,
+    response_model=Coverage | CoverageCollection,
     response_model_exclude_none=True,
 )
 async def get_data_location_id(
     location_id: str = Path(..., example="06260"),
-    parameter_name: str = Query(..., alias="parameter-name", example="dd,ff,rh,pp,tn"),
-    datetime: str | None = None,
+    parameter_name: str = Query(None, alias="parameter-name", example="dd,ff,rh,pp,tn"),
```

Comment:
I am not a big fan of this, we are basically saying that the type is `str`, but the default is `None`, which doesn't fit in that type.

Also, providing the default through `Query` like this is now deprecated. How about this?

```suggestion
    parameter_name: Annotated[str | None, Query(alias="parameter-name", example="dd,ff,rh,pp,tn")] = None,
):
```
See https://fastapi.tiangolo.com/tutorial/query-params-str-validations/#add-more-validations

Same in several other places.

Reply:
Refactored in https://github.com/EURODEO/e-soh/pull/30/commits/f1b1986323ad0b76c2eac6103f256ad90e201a91

Reply:
Thanks. Can we get all the `Query` ones? There is a bunch more, lines:
35, 69, 72, etc...

Reply:
Sorry, I misunderstood the first comment. Added Annotated to each `Query` and `Path` as recommended by FastAPI